### ISSUE TITLE
3.6.2: Fatal Error Thrown When HTMLDocument::query invoked but XPath not set up

### ISSUE THREAD

**Issue by hellofromtonya** [wp-media/wp-rocket#2865]
See ticket https://secure.helpscout.net/conversation/1220180136/179215/

A fatal error is thrown when XPath is not set up and `HTMLDocument::query` is invoked. 

```
PHP Fatal error: Uncaught Error: Call to a member function query() on bool in /var/www/x/public_html/x/wp-content/plugins/wp-rocket/inc/Engine/CriticalPath/AsyncCSS.php:134

Stack trace:
#0 /var/www/x/public_html/x/wp-content/plugins/wp-rocket/inc/Engine/CriticalPath/CriticalCSSSubscriber.php(591): WP_Rocket\\Engine\\CriticalPath\\AsyncCSS->modify_html()
#1 /var/www/x/public_html/x/wp-includes/class-wp-hook.php(287): WP_Rocket\\Engine\\CriticalPath\\CriticalCSSSubscriber->async_css()
#2 /var/www/x/public_html/x/wp-includes/plugin.php(206): WP_Hook->apply_filters()
#3 /var/www/x/public_html/x/wp-content/plugins/wp-rocket/x/classes/Buffer/class-optimization.php(104): apply_filters()
#4 [internal function]: WP_Rocket\\Buffer\\Optimization->maybe_process_buffer()
#5 /var/www/x/public_html/x/wp-includes/functions.php(4609): ob_end_flush()
#6 /var/www/hoke.org/public_html/john/wp-includes/class-wp-hook.php(287): wp_ob_end_flush_all()
#7 ...
```

This appears to happen if an empty string is passed to one of the named constructors. However, we should still guard against this to ensure no fatal error is thrown.

**Comment by hellofromtonya** [comment]
Fixed in PR 7 in the DOM Module repo https://github.com/wp-media/module-dom/pull/7

**Comment by NataliaDrause** [comment]
Ticket: https://secure.helpscout.net/conversation/1230381041/181347/

**Comment by arunbasillal** [comment]
@hellofromtonya Is this still a valid issue? It seems to be resolved with PR #2868

**Comment by hellofromtonya** [comment]
@arunbasillal No it's not. Closing it.
